pywebpack/__init__.py

```diff
@@ -7,7 +7,289 @@
 # it under the terms of the Revised BSD License; see LICENSE file for
 # more details.
 
-"""Webpack integration layer for Python."""
+r"""Webpack integration layer for Python.
+
+A simple example
+----------------
+
+Let's take a simple ``package.json`` that includes a Node module::
+
+    {
+      "private": true,
+      "name": "example",
+      "version": "0.0.1",
+      "author": "myself",
+      "license": "WTFPL",
+      "description": "example",
+      "dependencies": {
+        "jquery": "^3.2.1"
+      }
+    }
+
+And let's create a Webpack configuration file ``webpack.config.js`` to load the
+entry point ``src/index.js`` and output the built version in the ``dist``
+folder::
+
+    var path = require('path');
+
+    module.exports = {
+      context: path.resolve(__dirname, 'src'),
+      entry: './index.js',
+      output: {
+        filename: '[name].js',
+        path: path.resolve(__dirname, 'dist')
+      }
+    };
+
+Now, let's add to the ``packages.json`` a run script ``build`` that executes
+webpack::
+
+    "scripts": {
+      "build": "webpack --config webpack.config.js"
+    }
+
+We can easily wrap our project in a :class:`~pywebpack.project.WebpackProject`
+object::
+
+    from pywebpack.project import WebpackProject
+    project_path = '.'
+    project = WebpackProject(project_path)
+
+This will allow us to install the ``npm`` dependencies::
+
+    project.install()
+
+And invoke the ``npm`` ``build`` command to execute ``webpack`` to build the
+entry points::
+
+    project.build()
+
+Alternative, :meth:`~pywebpack.project.WebpackProject.buildall` can be used to
+execute both tasks at once.
+
+Build time config
+-----------------
+
+If we need to inject extra configuration at build time we can define a
+:class:`~pywebpack.project.WebpackTemplateProject`::
+
+    from pywebpack.bundle import WebpackTemplateProject
+    project = WebpackTemplateProject(
+        working_dir='tmp',  # where config and assets files will be copied
+        project_template_dir='buildconfig',  # `webpack.config.js` location
+        config={'debug': True},
+        config_path='build/config.json',
+    )
+
+``debug: True`` is an example of configuration that can be injected from Python
+and expose it to Webpack via the generated ``config.json``.
+
+Assets for multiple modules
+---------------------------
+
+It is a common practice to split your assets in several files to decrease
```

Comment:
"common practice" - this could be common practice for python but for webpack projects it's not common.

Perhaps here I would rather say that, 
- a) normally you just have a Webpack project with everything and you can happily live your life  
- b) for some advanced use cases (like a python webapp that allows plugins to distribute JS assets) you need to combine assets from the different python modules into a single webpack project, so you can build and optimize assets globally.